sdk/search/Azure.Search.Documents/src/Indexes/FieldBuilder.cs

```diff
@@ -27,6 +27,8 @@ public class FieldBuilder
                 new Dictionary<Type, SearchFieldDataType>()
                 {
                     [typeof(string)] = SearchFieldDataType.String,
+                    [typeof(sbyte)] = SearchFieldDataType.SByte,
+                    [typeof(short)] = SearchFieldDataType.Int16,
```

Comment:
@jairmyree FYI, we should do the same thing in Java for Int16, will need to look into if SByte can be supported

Reply:
Java has a concept of `byte` which is equivalent to SByte